File: linkedLists/reverseList/reverseList.js
// Given the head of a singly linked list,
// reverse the list, and return the reversed list.

// source: https://leetcode.com/problems/reverse-linked-list/


var reverseList = function(head) {
    let prev = null;
    let next = null;

    while (head) {
        next = head.next;
        head.next = prev;
        prev = head;
        head = next;
    }
    return prev;
};

File: linkedLists/reverseList/reverse_list.py
class Node:
    def __init__(self, val):
        self.val = val
        self.next = None

def reverse_list(head):
    prev = None
    curr = head

    while curr is not None:
        next = curr.next
        curr.next = prev
        prev = curr
        curr = next

    return prev
